### PR TITLE
feat: add experimental support for Cosign signing and verification

### PR DESCRIPTION
**Pull request by ningziwen**
https://github.com/runfinch/finch/issues/295

*Description of changes:*
Add COSIGN_PASSWORD env pass-through to allow users use Cosign.

The feature should be experimental as it is experimental in Nerdctl. As Finch points to Nerdctl documentation today, users could see Cosign feature is experimental in Nerdctl documentation so will not mention experimental in Finch explicitly.

*Testing done:*
The tests only covers signing by push and verification by pull and run with the keys as MVP. The tests won't pass until the [changed](https://github.com/containerd/nerdctl/pull/2109) is integrated to Finch. Tested locally that it can work with this change in Nerdctl.

- [ X ] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by pendo324** [comment]
LGTM, but need to update unit tests. There should also probably be an integration test for this new env pass-through, or Cosign itself

File: cmd/finch/nerdctl.go [hunk 1]
```diff
@@ -121,10 +122,27 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
 				arg = fmt.Sprintf("%s%s", arg[0:11], resolvedIP)
 			}
 			nerdctlArgs = append(nerdctlArgs, arg)
+		case arg == "--sign=cosign" || arg == "--verify=cosign":
```
Comment (davidhsingyuchen) [inline cmd/finch/nerdctl.go:125]: We would need to also add `"--sign cosign"` and `"--verify cosign"` if we go with the current approach. +1 on https://github.com/runfinch/finch/pull/296/files#r1142351036

File: cmd/finch/nerdctl.go [hunk 2]
```diff
@@ -121,10 +122,27 @@ func (nc *nerdctlCommand) run(cmdName string, args []string) error {
...
 		default:
 			nerdctlArgs = append(nerdctlArgs, arg)
 		}
 	}
+	limaArgs := []string{"shell", limaInstanceName, "sudo"}
+
+	// The env variables are not passed through by `limactl shell` by default so we need to assign the variables explicitly.
+	// https://github.com/lima-vm/lima/issues/1419
+	if cosign {
+		cosignEnv := "COSIGN_PASSWORD"
```
[the remark this answers is not in the document]
Comment (davidhsingyuchen) [reply cmd/finch/nerdctl.go:137]: +1. In our user doc, we may maintain a list of environment variables that are passed through the VM shell command, and `COSIGN_PASSWORD` would be one of them.

**Review by pendo324 — APPROVED** [review]

**Comment by ningziwen** [comment]
@pendo324 Changed the commit message to add experimental

**Pull request merged**